### PR TITLE
1759 mapping presets

### PR DESCRIPTION
**Pull request by adrian-lara**
This PR is ready for review, but note that once new user advice/text has been reviewed, **these texts need to be translated via lokalise**. The DO NOT MERGE label should probably be applied but I was concerned that this wouldn't be passed over for review because of it

**This PR contains migrations.**

#### Considerations/Notes:
- As a side-effect, this PR removes the old "incorrect" mapping history issue 1992 (not linked as it's not solved here). It is solved by this PR, #2002, which should both be deployed together.
- SEED-generated mapping suggestions and the corresponding flows (i.e. 'Upload Portfolio Manage data' button) are not being used anymore. The logic to generate and use these remain in the app for now in case parts of it do want to be used. This includes the ColumnMapping model which I recommend be deprecated once users have fully transitioned to using presets.

#### Any background context you want to provide?
Users wanted a greater level of consistency, customization, and reliability when mapping import file columns to SEED columns. Allowing them to maintain and apply their own presets is an attempt to provide this.

#### What's this PR do?
Remove old Column Mapping concept where a single set of Column Mappings is kept and continually updated for an organization by their most recent imports. This concept was replaced by the concept of Column Mapping **Presets**, where users of an org can define, maintain, and apply their own set of presets.

Each organization is given 2 default presets:
1. A snapshot of the current (old) Column Mappings (if applicable).
2. Default PM column mappings as defined in SEED (using https://github.com/SEED-platform/seed/blob/develop/seed/lib/mappings/data/pm-mapping.json)

Also, "Remove Column Mapping" button on Admin page is removed.

#### How should this be manually tested?
CRUD mapping presets in the old and now repurposed Column Mapping page.
CRU (not delete) mapping presets in the File Import process.
Go through any flows that users might encounter involving the above two.

#### What are the relevant tickets?
#1759 
#94 - removed no longer applicable documentation

### CODE REVIEW

**Comment by adrian-lara** [comment]
> Is it possible for the program to use the previous 'find suggested column mappings' when we paste in the headers into the box? Or did that functionality get completely removed?
> 
> image
> 
> That is use this method:
> 
> https://github.com/SEED-platform/seed/blob/32bf352bf090669ab76f0f8df76df44ba6ef3560/seed/lib/mcm/mapper.py#L26
> 
> .

**Did you envision the suggestions would pre-fill the SEED headers once "Copy into Data File headers" is clicked? or did you have something else in mind?**

The code still exists, but there's a few things to consider that could make it work slightly different. Finding suggested column mappings can be summarized by these lines of proposed deleted documentation: https://github.com/SEED-platform/seed/pull/2022/files#diff-df98e7e0427152ab152ef4cf76362fccL31-L37

**TL;DR - possible that only step 2c will be run.**

As of now, previous mappings are still in the app, but I was going to propose their removal once users have fully transitioned to the presets. If we wanted completely remove them from the app, that first step, 2a, would be skipped.

Also, step 2b, was only "triggered" when the PM specific upload was happening during the old file import process. I'm not sure if we'd want this triggered all the time? Or skipped completely? Or given it's own button on this page?

I'm happy to implement the suggestions into this page, but wanted to get your thoughts/reactions to the above.

**Comment by nllong** [comment]
What I’m thinking is to provide a button on the mapping profile page where the user can click and get the best matches based on all the column mappings that exist across all of their profiles. This includes best mapping to the known database fields. The reason why I want something like this is to allow the user to quickly get up and running when importing a new file... they still need to check the mappings, but at least they’ll have a good start. Does this make sense?

**Comment by adrian-lara** [comment]
Yeah that makes sense. Just to be sure, you're saying you essentially want to get suggestions based on only the previous step 2c? `...matches between raw and existing database columns using Jaro-Winkler distance...[for priority]`

**Comment by nllong** [comment]
> Yeah that makes sense. Just to be sure, you're saying you essentially want to get suggestions based on only the previous step 2c? `...matches between raw and existing database columns using Jaro-Winkler distance...[for priority]`

yep, exactly. And make it a button that the user clicks to return "best known matches".

**Comment by adrian-lara** [comment]
@nllong I definitely plan on adding the button to the Column Mapping Settings page (org settings), and am thinking it can't hurt to have on the actual import mapping page as well. 

Just wanted to confirm, that what you had in mind? Or would it somehow be "a little too much" customizability for the actual import mapping page maybe since that page was more for tweaks?

**Comment by adrian-lara** [comment]
> @nllong I definitely plan on adding the button to the Column Mapping Settings page (org settings), and am thinking it can't hurt to have on the actual import mapping page as well.
> 
> Just wanted to confirm, that what you had in mind? Or would it somehow be "a little too much" customizability for the actual import mapping page maybe since that page was more for tweaks?

@nllong following up on this and sharing the current iteration of the button. Let me know your thoughts on both. Thanks! 😄 
Screen Shot 2019-12-18 at 12 19 18 PM

**Comment by nllong** [comment]
> > @nllong I definitely plan on adding the button to the Column Mapping Settings page (org settings), and am thinking it can't hurt to have on the actual import mapping page as well.
> > Just wanted to confirm, that what you had in mind? Or would it somehow be "a little too much" customizability for the actual import mapping page maybe since that page was more for tweaks?
> 
> @nllong following up on this and sharing the current iteration of the button. Let me know your thoughts on both. Thanks! 😄
> Screen Shot 2019-12-18 at 12 19 18 PM

I think the screen shot. This is exactly what I had in mind.

File: seed/migrations/0116_columnmappingpreset.py [hunk 1]
```diff
@@ -71,7 +71,7 @@ class Migration(migrations.Migration):
 
     dependencies = [
         ('orgs', '0011_auto_20190714_2159'),
-        ('seed', '0114_auto_20191211_0958'),
+        ('seed', '0115_rehash_postal_code'),
```
Comment (nllong) [inline seed/migrations/0116_columnmappingpreset.py]: nice catch!

**Comment by nllong** [comment]
i think the translations are ready too. verbiage seems reasonable.

**Comment by nllong** [comment]
@adrian-lara can you update the django.mo files again, sorry 😢
